modules/navier_stokes/src/fvbcs/INSFVTurbulentTemperatureWallFunction.C

```diff
@@ -80,7 +83,9 @@ INSFVTurbulentTemperatureWallFunction::computeQpResidual()
     velocity(2) = (*_w_var)(current_argument, state);
 
   // Compute the velocity and direction of the velocity component that is parallel to the wall
-  const ADReal parallel_speed = (velocity - velocity * (fi.normal()) * (fi.normal())).norm();
+  const VectorValue offset = {1e-42, 1e-42, 1e-42};
+  const ADReal parallel_speed =
+      NS::computeSpeed(velocity - velocity * (fi.normal()) * (fi.normal()) + offset);
```

Comment:
Could we also add the nonlinear flag here? Offsetting the parallel velocity is not really needed for the linear solver and is adding unnecessary operations every time that we compute the wall function